api/core/request_fields.go

```diff
@@ -205,8 +207,17 @@ func checkRequestPostFiles(request *http.Request, endpoint string, apiRequest an
     postFiles, err := storeRequestFiles(request);
 
     if (err != nil) {
-        return NewBareInternalError("-029", endpoint, "Failed to store files from POST.").Err(err).
-                Add("struct-name", structName).Add("field-name", fieldType.Name);
+        var fileSizeExceededError *fileSizeExceededError
+        switch {
+            case errors.As(err, &fileSizeExceededError):
+                return NewBareBadRequestError("-036", endpoint, err.Error()).Err(err).
+                    Add("struct-name", structName).Add("field-name", fieldType.Name).
+                    Add("filename", fileSizeExceededError.Filename).Add("file-size", fileSizeExceededError.FileSizeKB).
+                    Add("max-file-size-kb", fileSizeExceededError.MaxFileSizeKB)
```

Comment:
I know you had me use a normal type switch here but I have two arguments for this method.  One, this allows easy access of the fields of the error.  And two, if the error is wrapped (which I realize it isn't now) it would always fail the type switch.  If something were to change for some reason and this error got wrapped in the future it might introduce a stealthy bug.  Maybe I am thinking too deeply on it and this is not the way to go about it.  Lemme know what you think.